frontend/config-assistant/src/components/code-editor/AceEditor.vue

```diff
@@ -83,45 +82,48 @@ onMounted(() => {
         sessionStore.lastChangeResponsible = ChangeResponsible.CodeEditor;
         const fileContentString = editor.value.getValue();
 
-        // Current workaround until schema of schema editor works: just accept schema without validation
-        //fileData.value = JSON.parse(jsonString);
-        if (sessionStore.currentMode === SessionMode.SchemaEditor) {
-          try {
-            fileData.value = manipulator.parseFileContent(fileContentString);
-          } catch (e) {
-            errorService.onErrorThrottled(e);
-          }
-          return;
-        }
-
         try {
           const parsedContent = manipulator.parseFileContent(fileContentString);
-          if (useSettingsStore().settingsData.codeEditor.allowSchemaViolatingInput) {
-            fileData.value = parsedContent;
-          }
-          validateAgainstSchemaThrottled(parsedContent)
-            .then(valid => {
-              if (valid) {
-                fileData.value = parsedContent;
-              } else {
-                // TODO: show error in editor
-                errorService.onWarningThrottled(new Error('Invalid JSON according to the schema.'));
-                //TODO: more detailed error message
-              }
-            })
-            .catch(e => errorService.onErrorThrottled(e));
+          fileData.value = parsedContent;
+
+          validateAgainstSchemaDebounced(parsedContent);
         } catch (e) {
-          // Do nothing: showing the parse error in the editor is enough
+          // if file content can not be parsed, that is because of error in input
+          // Invalid JSON is already highlighted by Ace Editor -> no action needed here
         }
       },
       WRITE_DEBOUNCE_TIME,
       {maxWait: 10 * WRITE_DEBOUNCE_TIME}
     )
   );
 
-  const validateAgainstSchemaThrottled = useThrottleFn((parsedContent: any) => {
-    return schemaValidationFunction.value(parsedContent);
-  }, SCHEMA_VALIDATION_THROTTLE_TIME);
+  const validateAgainstSchemaDebounced = useDebounceFn((parsedContent: any) => {
```

Comment:
This method got really big over the time and deeply nested, don't maybe we should split it up